src/MonitorProfiler/MainProfiler/MainProfiler.cpp

```diff
@@ -348,6 +370,27 @@ HRESULT STDMETHODCALLTYPE MainProfiler::GetReJITParameters(ModuleID moduleId, md
     {
         return m_pProbeInstrumentation->GetReJITParameters(moduleId, methodId, pFunctionControl);
     }
-    
+
+    return S_OK;
+}
+
+STDAPI DLLEXPORT RegisterMonitorMessageCallback(
+    ManagedMessageCallback pCallback)
+{
+    lock_guard<mutex> lock(g_managedMessageCallbackMutex);
```

Comment:
Add some comments here regarding why the locks are desired and what it's guarding